### PR TITLE
Initial version using newest SDK supporting WinForms & WPF

### LINKED ISSUE #4821: Designer code generation & csproj layout
I noticed that the resource files weren't updating the designer. I furthermore noticed this:
image

Changed the dropdown to `Public`, and the designer got updated. Furthermore, the `csproj` file got altered as shown:

https://github.com/rubberduck-vba/Rubberduck/blob/c22ae5eafb0c9ecde111e53146e76acc24b5fe57/Rubberduck.Resources/Rubberduck.Resources.csproj#L24-L45

Though in theory, globbing should be doing that, it's not correct. I modified the lines in the csproj as following:

```
  <ItemGroup>
    <EmbeddedResource Update="**\*.resx">
      <Generator>PublicResXFileCodeGenerator</Generator>
      <LastGenOutput>$([System.String]::Copy('%(FileName)')).Designer.cs</LastGenOutput>
    </EmbeddedResource>
    <Compile Update="**\*.Designer.cs">
      <DesignTime>True</DesignTime>
      <AutoGen>True</AutoGen>
      <DependentUpon>$([System.String]::Copy('%(Filename)').Replace('.Designer', '')).resx</DependentUpon>
    </Compile>
  </ItemGroup>
```

This seems to work but has a side effect: Whenever any changes are made, the csproj get updated with the hard-coded version automatically:

```
  <ItemGroup>
    <Compile Update="Inspections\InspectionNames.Designer.cs">
      <DesignTime>True</DesignTime>
      <AutoGen>True</AutoGen>
      <DependentUpon>InspectionNames.resx</DependentUpon>
    </Compile>
  </ItemGroup>
  <ItemGroup>
    <EmbeddedResource Update="Inspections\InspectionNames.resx">
      <Generator>PublicResXFileCodeGenerator</Generator>
      <LastGenOutput>InspectionNames.Designer.cs</LastGenOutput>
    </EmbeddedResource>
  </ItemGroup>
```

I am not clear if globbing is working at all, but I do think that the csproj will still end up writing out the explicit directives once we enable designer. It might be even possible that this will be the case as long we stay on 2017. As a result, this makes the csproj less then clean when building ([ref](https://github.com/rubberduck-vba/Rubberduck/pull/4820/files#diff-cedee1d341f4a81ef416a7add5ee43c1)).

We can choose to allow csproj to have its own way -- it will be automatically managed, and it would be only applicable to a subset of file types (in this particular case, the resource files) which we aren't churning out new ones on a daily basis ;-).

### LINKED ISSUE #4872: This repo fails to build/load in Visual Studio 2019
I got a report of a failure to build this repo, so I wanted to make y'all aware of the problem.

If you try to load `Rubberduck.sln` (or any individual project) in Visual Studio 2019 (Preview), it will fail. This is because `Sunburst.NET.Sdk.WPF/1.0.47` isn't compatible with 2019. From SunburstApps/MSBuildSdks#1, it sounds like Rubberduck should consider migrating to a different approach.

As mentioned there, one option is to wait for .NET Core 3.0 and its associated SDK, which [will support WPF projects](https://devblogs.microsoft.com/dotnet/net-core-3-and-support-for-windows-desktop-applications/). But that's not yet released.

### PR DESCRIPTION
**Pull request by bclothier**
Closes #4872 
Closes #4821

NOTE: Do not merge PR until end of the month.

Merging the PR will mean that we no longer can use Visual Studio 2017; all contributors must start using Visual Studio 2019 after the merging. As a benefit, we are able to remove all the hacky workarounds that was put in place to support WinForms/WPF when we migrated to SDK-style project format. However, we need to test that we are able to build on AV correctly.

### CODE REVIEW

**Comment by IvenBach** [comment]
If this PR will invalidate any Visual Studio prior to 2019 it may be prudent to give a couple weeks notice before merging. I know I'd be at a loss right now as I am still on VS2017 and am really out of the RD development loop.

**Comment by bclothier** [comment]
The earliest it can merge would be Nov 1st, and even so, I don't know it'll merge that early. More likely, we'll want to do that after our green release, which is likely to be in mid-November anyway.

Build articles will be updated as well, so you can consider this PR your advance warning. :)

**Comment by bclothier** [comment]
Both designers individually seem to work but there are issues with loading WinForm designer with WPF element host, because WPF can't find itself, apparently. However, it is verified that runtime does work just fine. 

There is still the bogus warning about re-importing SDK because the `Core` and `Main` uses a different SDK than the rest of projects. To handle that, conditions may be needed. and will be investigated.

**Comment by Vogel612** [comment]
for the record: I can now try to run a build on my linux machine. Understandably it utterly fails because we're still targeting .NETFramework, but this is going to go a long way to enable development on other platforms, possibly allowing us to reuse parts of Rubberduck in something like a VSCode plugin or a cross-platform standalone editor :)

**Comment by mansellan** [comment]
Perhaps we should soon consider targetting [Net Standard](https://docs.microsoft.com/en-us/dotnet/standard/net-standard) for most of the RD projects. Ideally we'd target 2.0, but that requires net472 for full compatibility (see the note in the table on the linked page). Alternatively we could target 1.4, which maps directly to net461 (which we will soon need to move to anyway) - it depends which features we're using.

**Comment by bclothier** [comment]
After doing some experimentation, I'm convinced that Rubberduck will not be leaving framework anytime soon. Even targeting .NET Standard 1.4 for some of the projects will not work due to the fact that meta projects cannot run using .NET Standard 1.4 and there are numerous gaps in the .NET Core 3.0 as well.

**Comment by bclothier** [comment]
This branch is borked. Going to make a new PR.

**Comment by bclothier** [comment]
Though this PR is ready for merging, we probably should hold off until end of the month. Nonetheless, it'll be good to review this now that this is finally ready.

File: Rubberduck.UnitTesting/Properties/UnitTestDefaults.Designer.cs [hunk 1]
```diff
@@ -12,7 +12,7 @@ namespace Rubberduck.UnitTesting.Properties {
     
     
     [global::System.Runtime.CompilerServices.CompilerGeneratedAttribute()]
-    [global::System.CodeDom.Compiler.GeneratedCodeAttribute("Microsoft.VisualStudio.Editors.SettingsDesigner.SettingsSingleFileGenerator", "15.9.0.0")]
+    [global::System.CodeDom.Compiler.GeneratedCodeAttribute("Microsoft.VisualStudio.Editors.SettingsDesigner.SettingsSingleFileGenerator", "16.5.0.0")]
```
Comment (Vogel612) [inline Rubberduck.UnitTesting/Properties/UnitTestDefaults.Designer.cs:15]: interesting...why did that not change with all the other Designers?

File: appveyor.yml [hunk 2]
```diff
@@ -30,7 +30,7 @@ branches:
 skip_tags: true
 
 # build-matrix: all Release CPUs on VS2017
-image: Visual Studio 2017
+image: Visual Studio 2019
```
Comment (Vogel612) [inline appveyor.yml:33]: comment is now a liar :)

**Pull request merged**